### ISSUE TITLE
MethodNotFound exception thrown when batching files

### ISSUE THREAD

**Issue by kcootedinh** [Pablissimo/SonarTsPlugin#49]
I'm having this issue when using the plugin. It seems to be when there are enough files that it causes the plugin to require batching. When I limit the sonar.sources to not require batching then it seems to work. The exception is thrown after the last batch is run.

I've tried versions 0.8 and 0.9.

Running with 0.8 (just happened to be the last version I attempted with) the stack trace is below. When running with 0.9 the exception is the same and is thrown from the same method.

```
org.sonar.runner.impl.RunnerException: Unable to execute Sonar
        at org.sonar.runner.impl.BatchLauncher$1.delegateExecution(BatchLauncher.java:91)
        at org.sonar.runner.impl.BatchLauncher$1.run(BatchLauncher.java:75)
        at java.security.AccessController.doPrivileged(Native Method)
        at org.sonar.runner.impl.BatchLauncher.doExecute(BatchLauncher.java:69)
        at org.sonar.runner.impl.BatchLauncher.execute(BatchLauncher.java:50)
        at org.sonar.runner.api.EmbeddedRunner.doExecute(EmbeddedRunner.java:102)
        at org.sonar.runner.api.Runner.execute(Runner.java:100)
        at org.sonar.runner.Main.executeTask(Main.java:70)
        at org.sonar.runner.Main.execute(Main.java:59)
        at org.sonar.runner.Main.main(Main.java:53)
Caused by: java.lang.NoSuchMethodError: org.sonar.api.resources.File.fromIOFile(Ljava/io/File;Lorg/sonar/api/resources/Project;)Lorg/sonar/api/resources/File;
        at com.pablissimo.sonar.TsLintSensor.getFileFromIOFile(TsLintSensor.java:168)
        at com.pablissimo.sonar.TsLintSensor.analyse(TsLintSensor.java:143)
        at org.sonar.scanner.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:57)
        at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:49)
        at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:78)
        at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:184)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:142)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:127)
        at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:241)
        at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:236)
        at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:226)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:142)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:127)
        at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:47)
        at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:86)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:142)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:127)
        at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:115)
        at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:118)
        at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:78)
        at org.sonar.runner.batch.IsolatedLauncher.execute(IsolatedLauncher.java:48)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.sonar.runner.impl.BatchLauncher$1.delegateExecution(BatchLauncher.java:87)
        ... 9 more
```

**Comment by Pablissimo** [comment]
What version of the runner are you using? The missing method looks like its in the SonarQube API and I admit this isn't something I've seen before!

**Comment by Pablissimo** [comment]
(For reference, I test locally with 2.5 but notice that 2.6 is available which I've yet to touch)

**Comment by Pablissimo** [comment]
(Also, and sorry for the parantheticals, what version of SonarQube please?)

**Comment by kcootedinh** [comment]
Version of the runner is 2.4.1 and version of SonarQube is 6.0. 

I have a feeling I need to update the runner and see how I go.

**Comment by Pablissimo** [comment]
I've not tested on SonarQube 6.0 actually, I wonder if it's that side of things. I'll try against 6.0 and see what happens with the 2.5 runner to eliminate that side of things.

**Comment by Pablissimo** [comment]
Yeah looks like I'm not compatible with SonarQube 6.0 - will need to make a change to support it but it's unclear if I can do it in a backwards-compatible way. Shall research!

**Comment by kcootedinh** [comment]
Fair enough! Thanks for the prompt response.

**Comment by Pablissimo** [comment]
Yeah I've been targeting a far-too-old version of the API and some bits have gone from deprecated to just gone. Started the process of updating to the latest API (will limit use of the plugin to SonarQube 5.6+ but that's fine) but it's a bit of a faff, will advise when I have a beta if you're interested in giving it a go?

**Comment by kcootedinh** [comment]
Yeah definitely

**Comment by Pablissimo** [comment]
Ok, I've got a compiling-and-maybe-sort-of-working version going on SQ 6.0, but am still fixing unit tests and have yet to check for regressions (of which there will be some). But if you're feeling brave then I should have something uploaded this afternoon.

**Comment by kcootedinh** [comment]
No rush, it's Saturday night on a long weekend here. I'll have to wait until Tuesday until I can test it.

**Comment by Eulinky** [comment]
Nice! Just started a two weeks holidays so take your time! Looking forward
to check out the plugin, our team will love it!

Am 01.10.2016 11:51 vorm. schrieb "Pablissimo" <notifications@github.com>:

Ok, I've got a compiling-and-maybe-sort-of-working version going on SQ 6.0,
but am still fixing unit tests and have yet to check for regressions (of
which there will be some). But if you're feeling brave then I should have
something uploaded this afternoon.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/Pablissimo/SonarTsPlugin/issues/49#issuecomment-250903744>,
or mute the thread

**Comment by Pablissimo** [comment]
Ok, finally a release but caveats ahoy - the source for this remains on the 'sq6support' branch but will get merged to master once people start feeling comfortable that everything isn't broken.

https://github.com/Pablissimo/SonarTsPlugin/releases/tag/v0.93

I can't stress enough how preliminary SonarQube 6.0 support is, and that I've almost certainly regressed something in reworking things to support the updated API but locally on a few test cases on both Windows and Linux it appears to broadly work as it did before. Your mileage may vary etc...

**Comment by kcootedinh** [comment]
The plugin was updated and is being used and seems to be working.

**Comment by Pablissimo** [comment]
Fixed in v0.94.
